### DISCUSSION TITLE
Browse Mode (Open file as read only)

### DISCUSSION THREAD

**Discussion by Sdthedawson** [codefori/vscode-ibmi#1048] [Q&A] [answered]
It would be really cool if there was a feature similar to RDI that allowed you to open a file in Browse Mode ie. open it as read only.
![image](https://user-images.githubusercontent.com/1506998/217632922-a70c58ab-d6fb-42cb-b332-e41692d368fe.png)

**Comment by worksofliam** [comment, marked as answer]
@Sdthedawson Great question!

I am afraid we don't support readonly mode. This is a limitation of VS Code.

**Reply by Sdthedawson** [reply]
Thank you for the fast response!
Ill put in a request to VS Code in that case! Lol

**Reply by sebjulliand** [reply]
> @Sdthedawson Great question!
> 
> I am afraid we don't support readonly mode. This is a limitation of VS Code.

@worksofliam I guess that technically, we could have an option on QSYS/IFS filters to mark them as "read only filters" and prevent saving on members/files listed by these filters. That's not an actual read only mode but that the closest possible solution that comes to my mind 😅

**Reply by Sdthedawson** [reply]
That would work! Would there be anyway to apply that to the member open with CTLR-ALT-O as well?

**Reply by sebjulliand** [reply]
I think so 🙂
@worksofliam if you think it's worth a shot, I'll be happy to make that PR.

**Reply by worksofliam** [reply]
@sebjulliand go for it mate! I will be waiting curiously!

**Comment by sebjulliand** [comment]
@Sdthedawson , PR https://github.com/halcyon-tech/vscode-ibmi/pull/1075 has been merged, you can expect browsing capabilities in the next release.

**Reply by Sdthedawson** [reply]
Thats awesome thank you so much!

How can i buy you a coffee?
